preprocessing/data_transformer.py

```diff
@@ -213,3 +212,80 @@ def encode_categorical_features(dataframe_dict: dict, columns_list: list,
         pd.set_option('display.max_rows', None)
 
     return dataframe_dict
+
+
+def filter_category(value: str,
+                    to_remove: set) -> str:
+    """ Category filter
+
+    This function resets the value to NAN it it's in corresponding set
+
+    :param str value: a value to be checked.
+    :param set to_remove: a set of values to be reset.
+
+    :return:
+            value: an original value or the NAN-value.
+    """
+    if value in to_remove:
+        return np.nan
+    return value
+
+
+def clean_categorical_features(dataframe_dict: object,
+                               columns_list: object,
+                               inform_only: object = True,
+                               print_results: object = True) -> object:
+    """ Categorical features cleaner
+
+    This function resets the value of a categorical feature to NAN is case it's not exists in both (training and test) datasets.
+    Currently only supports `len(dataframe_dict)=2`: train and test
+
+    :param dict dataframe_dict: a dictionary of Pandas dataframes.
+    :param list columns_list: The list of categorical features to be cleared.
+    :param bool inform_only: If True (default), no data is replaced -  just informing.
+    :param bool print_results: If False, no data is printed to the console. If True, all data is printed to the console.
+
+    :return:
+            dataframe_dict: a dictionary of Pandas dataframes after encoding.
+    """
+
+    # Check if it only one dataframe provided
+    if len(dataframe_dict) != 2:
+        # do nothing and return the original data
+        return dataframe_dict
```

Comment:
I would suggest providing a print statement here for telling the user that this function is not applied and the reason for this e.g.
print("Clean function can't be applied because the number of datasets is not equal to 2")
Moreover, it is a good idea to add some logs here as info e.g.
logger.info("Clean function can't be applied")